{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "host = 'http://143.248.23.236:4042/'\n",
    "route = \"maxentscan\"\n",
    "data ={\"seq\": \"NM_000368.5(TSC1):c.364-1G>T\", \"session_data\": \"kdw59520@kaist.ac.kr\"}\n",
    "response = requests.post(\n",
    "    url=f'{host}{route}',\n",
    "    json=data,\n",
    "    headers={\"Content-Type\": \"application/json; charset=UTF-8\"}\n",
    ")\n",
    "response_data = response.json()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "maxentscan_with_strand = {}\n",
    "for strand in response_data[\"resources\"].keys():\n",
    "    maxentscan_with_strand[strand] = { key: value for item in response_data[\"resources\"][strand] for key, value in [tuple(item.values())]}\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "host = 'http://143.248.208.160:4042/'\n",
    "route = \"parse2mane\"\n",
    "seq=\"chr1:925952-925952 G>A\"\n",
    "#seq=\"chrX:18609570-18609570 G>A\"\n",
    "#seq=\"NM_000368.5(TSC1):c.364-1G>T\"\n",
    "data ={\"seq\": seq, \"session_data\": \"kdw59520@kaist.ac.kr\"}\n",
    "response = requests.post(\n",
    "    url=f'{host}{route}',\n",
    "    json=data,\n",
    "    headers={\"Content-Type\": \"application/json; charset=UTF-8\"}\n",
    ")\n",
    "response_data = response.json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'resources': [{'Gene': 'SAMD11<br><a href=https://www.genecards.org/cgi-bin/carddisp.pl?gene=SAMD11 target=\"_blank\">GeneCards</a><br><a href=https://gtexportal.org/home/gene/SAMD11 target=\"_blank\">GTEx</a>',\n",
       "   'HGVS': '<a href=https://varsome.com/variant/hg38/ENST00000616016.5%3Ac.548G%3EA?annotation-mode=germline target=\"_blank\">(ENST00000616016.5):c.548G>A</a><br>(p.Gly183Glu)<br><br><a href=https://varsome.com/variant/hg38/NM_001385641.1%3Ac.548G%3EA?annotation-mode=germline target=\"_blank\">(NM_001385641.1):c.548G>A</a><br>(p.Gly183Glu)',\n",
       "   'Strand': '+ (plus strand)',\n",
       "   'Transcript ID': 'ENST00000616016.5<br><br>NM_001385641.1(refseq)',\n",
       "   'Variant': 'chr1:925952-925952 G>A<br><a href=http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr1%3A925952%2D925952&hgsid=1156735329_9yjwvrj3bzbN4iAJuhQElyCAtvvR target =\"_blank\">UCSC genome browser</a><br><a href=https://ensembl.org/Homo_sapiens/Location/View?r=1%3A925952-925952 target=\"_blank\">Ensembl genome browser</a><br><a href=\"https://mastermind.genomenon.com/detail?gene=SAMD11&mutation=c.548G>A&disease=all%20diseases\" target=\"_blank\">Mastermind</a>',\n",
       "   'Variant Type': '\\n                                <button class=\"button-consequence\" style=\"background-color: #ffd700;\" onclick=\"this.focus();\">\\n                                    Missense variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> MODERATE<br>\\n                                        <strong>Description:</strong> A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved\\n                                    </div>\\n                                </button>\\n                                ',\n",
       "   'ex_variants': 'rs1640863258'}]}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'so_term': 'transcript_ablation',\n",
       " 'so_description': 'A feature ablation whereby the deleted region includes a transcript feature',\n",
       " 'so_accession': 'SO:0001893',\n",
       " 'display_term': 'Transcript ablation',\n",
       " 'impact': 'HIGH',\n",
       " 'color': '#ff0000'}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import json\n",
    "import pprint\n",
    "consequence_data_path =\"../ensembl_variant_consequences.json\"\n",
    "with open(consequence_data_path) as f:\n",
    "    consequence_datas=json.load(f)\n",
    "consequence_datas[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "for item in response_data[\"resources\"]:\n",
    "    variant_types = item[\"Variant Type\"].split(\"&\")\n",
    "    variant_types_array = []\n",
    "    for variant_type in variant_types:\n",
    "        for consequence_data in consequence_datas:\n",
    "            if variant_type == consequence_data[\"so_term\"]:\n",
    "                button_html = f\"\"\"\n",
    "                            <button class=\"button\" style=\"background-color: {consequence_data['color']};\" onclick=\"this.focus();\">\n",
    "                                { consequence_data['display_term'] }\n",
    "                                <div class=\"tooltip-consequence\">\n",
    "                                    <strong>Impact:</strong> {consequence_data['impact'] }<br>\n",
    "                                    <strong>Description:</strong> { consequence_data['so_description'] }\n",
    "                                </div>\n",
    "                            </button>\n",
    "                            \"\"\"\n",
    "                #print(button_html)\n",
    "                variant_types_array.append(button_html)\n",
    "    join_button_html = \"<br>\".join(variant_types_array)\n",
    "    print(join_button_html)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "host = 'http://143.248.23.236:4042/'\n",
    "route = \"split2vcf\"\n",
    "seq=\"chrX:18609570-18609570 G>A\"\n",
    "#seq=\"NM_000368.5(TSC1):c.364-1G>T\"\n",
    "data ={\"seq\": seq, \"session_data\": \"kdw59520@kaist.ac.kr\"}\n",
    "response = requests.post(\n",
    "    url=f'{host}{route}',\n",
    "    json=data,\n",
    "    headers={\"Content-Type\": \"application/json; charset=UTF-8\"}\n",
    ")\n",
    "response_data = response.json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'APPRIS': '-',\n",
       "  'Canonical': '-',\n",
       "  'Clinical Significance': 'pathogenic&likely_pathogenic',\n",
       "  'Existing Variation': '<a href=https://www.ncbi.nlm.nih.gov/snp/rs267608653&CM085312?horizontal_tab=true target=\"_blank\">rs267608653&CM085312</a>',\n",
       "  'Exon': '15/22',\n",
       "  'Gene': 'CDKL5',\n",
       "  'Gene ID': 'ENSG00000008086',\n",
       "  'HGVSc': 'ENST00000379989.6:c.2152G>A',\n",
       "  'HGVSp': 'ENSP00000369325.3:p.Val718Met',\n",
       "  'Intron': '-',\n",
       "  'MANE Select': '-',\n",
       "  'Mane Plus Clinical': '-',\n",
       "  'PolyPhen2': 'probably_damaging(0.994)',\n",
       "  'REVEL': 'Pathogenic(0.625)',\n",
       "  'SIFT': 'deleterious_low_confidence(0)',\n",
       "  'SIFT4G': '-',\n",
       "  'Strand': '+',\n",
       "  'Transcript ID': 'ENST00000379989.6',\n",
       "  'Variant Type': '\\n                                <button class=\"button-consequence\" style=\"background-color: #ffd700;\" onclick=\"this.focus();\">\\n                                    Missense variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> MODERATE<br>\\n                                        <strong>Description:</strong> A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved\\n                                    </div>\\n                                </button>\\n                                \\n                                <button class=\"button-consequence\" style=\"background-color: #ff7f50;\" onclick=\"this.focus();\">\\n                                    Splice region variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> LOW<br>\\n                                        <strong>Description:</strong> A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron\\n                                    </div>\\n                                </button>\\n                                '},\n",
       " {'APPRIS': '-',\n",
       "  'Canonical': '-',\n",
       "  'Clinical Significance': 'pathogenic&likely_pathogenic',\n",
       "  'Existing Variation': '<a href=https://www.ncbi.nlm.nih.gov/snp/rs267608653&CM085312?horizontal_tab=true target=\"_blank\">rs267608653&CM085312</a>',\n",
       "  'Exon': '14/21',\n",
       "  'Gene': 'CDKL5',\n",
       "  'Gene ID': 'ENSG00000008086',\n",
       "  'HGVSc': 'ENST00000379996.7:c.2152G>A',\n",
       "  'HGVSp': 'ENSP00000369332.3:p.Val718Met',\n",
       "  'Intron': '-',\n",
       "  'MANE Select': '-',\n",
       "  'Mane Plus Clinical': '-',\n",
       "  'PolyPhen2': 'probably_damaging(0.994)',\n",
       "  'REVEL': 'Pathogenic(0.625)',\n",
       "  'SIFT': 'deleterious_low_confidence(0)',\n",
       "  'SIFT4G': '-',\n",
       "  'Strand': '+',\n",
       "  'Transcript ID': 'ENST00000379996.7',\n",
       "  'Variant Type': '\\n                                <button class=\"button-consequence\" style=\"background-color: #ffd700;\" onclick=\"this.focus();\">\\n                                    Missense variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> MODERATE<br>\\n                                        <strong>Description:</strong> A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved\\n                                    </div>\\n                                </button>\\n                                \\n                                <button class=\"button-consequence\" style=\"background-color: #ff7f50;\" onclick=\"this.focus();\">\\n                                    Splice region variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> LOW<br>\\n                                        <strong>Description:</strong> A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron\\n                                    </div>\\n                                </button>\\n                                '},\n",
       " {'APPRIS': '-',\n",
       "  'Canonical': '-',\n",
       "  'Clinical Significance': 'pathogenic&likely_pathogenic',\n",
       "  'Existing Variation': '<a href=https://www.ncbi.nlm.nih.gov/snp/rs267608653&CM085312?horizontal_tab=true target=\"_blank\">rs267608653&CM085312</a>',\n",
       "  'Exon': '14/16',\n",
       "  'Gene': 'CDKL5',\n",
       "  'Gene ID': 'ENSG00000008086',\n",
       "  'HGVSc': 'ENST00000463994.4:c.2152G>A',\n",
       "  'HGVSp': 'ENSP00000485184.1:p.Val718Met',\n",
       "  'Intron': '-',\n",
       "  'MANE Select': '-',\n",
       "  'Mane Plus Clinical': '-',\n",
       "  'PolyPhen2': 'probably_damaging(0.994)',\n",
       "  'REVEL': '-',\n",
       "  'SIFT': 'deleterious_low_confidence(0)',\n",
       "  'SIFT4G': '-',\n",
       "  'Strand': '+',\n",
       "  'Transcript ID': 'ENST00000463994.4',\n",
       "  'Variant Type': '\\n                                <button class=\"button-consequence\" style=\"background-color: #ffd700;\" onclick=\"this.focus();\">\\n                                    Missense variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> MODERATE<br>\\n                                        <strong>Description:</strong> A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved\\n                                    </div>\\n                                </button>\\n                                \\n                                <button class=\"button-consequence\" style=\"background-color: #ff7f50;\" onclick=\"this.focus();\">\\n                                    Splice region variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> LOW<br>\\n                                        <strong>Description:</strong> A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron\\n                                    </div>\\n                                </button>\\n                                '},\n",
       " {'APPRIS': 'P1',\n",
       "  'Canonical': 'YES',\n",
       "  'Clinical Significance': 'pathogenic&likely_pathogenic',\n",
       "  'Existing Variation': '<a href=https://www.ncbi.nlm.nih.gov/snp/rs267608653&CM085312?horizontal_tab=true target=\"_blank\">rs267608653&CM085312</a>',\n",
       "  'Exon': '14/18',\n",
       "  'Gene': 'CDKL5',\n",
       "  'Gene ID': 'ENSG00000008086',\n",
       "  'HGVSc': 'ENST00000623535.2:c.2152G>A',\n",
       "  'HGVSp': 'ENSP00000485244.1:p.Val718Met',\n",
       "  'Intron': '-',\n",
       "  'MANE Select': 'NM_001323289.2',\n",
       "  'Mane Plus Clinical': '-',\n",
       "  'PolyPhen2': 'probably_damaging(0.999)',\n",
       "  'REVEL': '-',\n",
       "  'SIFT': 'deleterious_low_confidence(0)',\n",
       "  'SIFT4G': '-',\n",
       "  'Strand': '+',\n",
       "  'Transcript ID': 'ENST00000623535.2',\n",
       "  'Variant Type': '\\n                                <button class=\"button-consequence\" style=\"background-color: #ffd700;\" onclick=\"this.focus();\">\\n                                    Missense variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> MODERATE<br>\\n                                        <strong>Description:</strong> A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved\\n                                    </div>\\n                                </button>\\n                                \\n                                <button class=\"button-consequence\" style=\"background-color: #ff7f50;\" onclick=\"this.focus();\">\\n                                    Splice region variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> LOW<br>\\n                                        <strong>Description:</strong> A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron\\n                                    </div>\\n                                </button>\\n                                '},\n",
       " {'APPRIS': '-',\n",
       "  'Canonical': '-',\n",
       "  'Clinical Significance': 'pathogenic&likely_pathogenic',\n",
       "  'Existing Variation': '<a href=https://www.ncbi.nlm.nih.gov/snp/rs267608653&CM085312?horizontal_tab=true target=\"_blank\">rs267608653&CM085312</a>',\n",
       "  'Exon': '14/18',\n",
       "  'Gene': 'CDKL5',\n",
       "  'Gene ID': 'ENSG00000008086',\n",
       "  'HGVSc': 'ENST00000635828.1:c.2152G>A',\n",
       "  'HGVSp': 'ENSP00000490170.1:p.Val718Met',\n",
       "  'Intron': '-',\n",
       "  'MANE Select': '-',\n",
       "  'Mane Plus Clinical': '-',\n",
       "  'PolyPhen2': 'probably_damaging(0.999)',\n",
       "  'REVEL': '-',\n",
       "  'SIFT': 'deleterious_low_confidence(0)',\n",
       "  'SIFT4G': '-',\n",
       "  'Strand': '+',\n",
       "  'Transcript ID': 'ENST00000635828.1',\n",
       "  'Variant Type': '\\n                                <button class=\"button-consequence\" style=\"background-color: #ffd700;\" onclick=\"this.focus();\">\\n                                    Missense variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> MODERATE<br>\\n                                        <strong>Description:</strong> A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved\\n                                    </div>\\n                                </button>\\n                                \\n                                <button class=\"button-consequence\" style=\"background-color: #ff7f50;\" onclick=\"this.focus();\">\\n                                    Splice region variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> LOW<br>\\n                                        <strong>Description:</strong> A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron\\n                                    </div>\\n                                </button>\\n                                '},\n",
       " {'APPRIS': '-',\n",
       "  'Canonical': '-',\n",
       "  'Clinical Significance': 'pathogenic&likely_pathogenic',\n",
       "  'Existing Variation': '<a href=https://www.ncbi.nlm.nih.gov/snp/rs267608653&CM085312?horizontal_tab=true target=\"_blank\">rs267608653&CM085312</a>',\n",
       "  'Exon': '14/19',\n",
       "  'Gene': 'CDKL5',\n",
       "  'Gene ID': 'ENSG00000008086',\n",
       "  'HGVSc': 'ENST00000674046.1:c.2152G>A',\n",
       "  'HGVSp': 'ENSP00000501174.1:p.Val718Met',\n",
       "  'Intron': '-',\n",
       "  'MANE Select': '-',\n",
       "  'Mane Plus Clinical': '-',\n",
       "  'PolyPhen2': 'probably_damaging(0.999)',\n",
       "  'REVEL': '-',\n",
       "  'SIFT': 'deleterious_low_confidence(0)',\n",
       "  'SIFT4G': '-',\n",
       "  'Strand': '+',\n",
       "  'Transcript ID': 'ENST00000674046.1',\n",
       "  'Variant Type': '\\n                                <button class=\"button-consequence\" style=\"background-color: #ffd700;\" onclick=\"this.focus();\">\\n                                    Missense variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> MODERATE<br>\\n                                        <strong>Description:</strong> A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved\\n                                    </div>\\n                                </button>\\n                                \\n                                <button class=\"button-consequence\" style=\"background-color: #ff7f50;\" onclick=\"this.focus();\">\\n                                    Splice region variant\\n                                    <div class=\"tooltip-consequence\">\\n                                        <strong>Impact:</strong> LOW<br>\\n                                        <strong>Description:</strong> A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron\\n                                    </div>\\n                                </button>\\n                                '}]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response_data['ensembl']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'accessions': ['litvar@rs121913527##'],\n",
      " 'count': 775,\n",
      " 'current': 2,\n",
      " 'facets': {'count': 775,\n",
      "            'journal': {'buckets': [{'count': 59, 'val': 'Oncotarget'},\n",
      "                                    {'count': 46, 'val': 'Cancers (Basel)'},\n",
      "                                    {'count': 46, 'val': 'PLoS One'},\n",
      "                                    {'count': 34, 'val': 'Nat Commun'},\n",
      "                                    {'count': 24, 'val': 'Front Oncol'}]},\n",
      "            'section': {'count': 870,\n",
      "                        'snippets': {'buckets': [{'count': 276,\n",
      "                                                  'cp': 276,\n",
      "                                                  'val': 'table'},\n",
      "                                                 {'count': 214,\n",
      "                                                  'cp': 214,\n",
      "                                                  'val': 'results'},\n",
      "                                                 {'count': 99,\n",
      "                                                  'cp': 99,\n",
      "                                                  'val': 'discussion'},\n",
      "                                                 {'count': 81,\n",
      "                                                  'cp': 81,\n",
      "                                                  'val': 'other'},\n",
      "                                                 {'count': 66,\n",
      "                                                  'cp': 66,\n",
      "                                                  'val': 'methods'},\n",
      "                                                 {'count': 50,\n",
      "                                                  'cp': 50,\n",
      "                                                  'val': 'figure_legend'},\n",
      "                                                 {'count': 48,\n",
      "                                                  'cp': 48,\n",
      "                                                  'val': 'abstract'},\n",
      "                                                 {'count': 30,\n",
      "                                                  'cp': 30,\n",
      "                                                  'val': 'introduction'},\n",
      "                                                 {'count': 4,\n",
      "                                                  'cp': 4,\n",
      "                                                  'val': 'conclusion'},\n",
      "                                                 {'count': 2,\n",
      "                                                  'cp': 2,\n",
      "                                                  'val': 'title'}]}},\n",
      "            'year': {'buckets': [{'count': 97, 'val': 2022},\n",
      "                                 {'count': 85, 'val': 2019},\n",
      "                                 {'count': 83, 'val': 2020},\n",
      "                                 {'count': 77, 'val': 2018},\n",
      "                                 {'count': 76, 'val': 2021},\n",
      "                                 {'count': 68, 'val': 2023},\n",
      "                                 {'count': 67, 'val': 2017},\n",
      "                                 {'count': 57, 'val': 2016},\n",
      "                                 {'count': 55, 'val': 2015},\n",
      "                                 {'count': 44, 'val': 2024},\n",
      "                                 {'count': 30, 'val': 2014},\n",
      "                                 {'count': 16, 'val': 2013},\n",
      "                                 {'count': 7, 'val': 2012},\n",
      "                                 {'count': 5, 'val': 2011},\n",
      "                                 {'count': 4, 'val': 2010},\n",
      "                                 {'count': 3, 'val': 2009},\n",
      "                                 {'count': 1, 'val': 2006}]}},\n",
      " 'page_size': 10,\n",
      " 'results': [{'authors': ['Zhang X',\n",
      "                          'Ma H',\n",
      "                          'He Y',\n",
      "                          'He W',\n",
      "                          'Chen N',\n",
      "                          'Li Y',\n",
      "                          'Zhong W',\n",
      "                          'Wu G',\n",
      "                          'Zhou X',\n",
      "                          'Hua H',\n",
      "                          'Ye F',\n",
      "                          'Cai H',\n",
      "                          'Jiang W'],\n",
      "              'citations': {'BibTeX': '@article{38293276, '\n",
      "                                      'title={Reclassification of RAS/BRAF '\n",
      "                                      'allele mutations predicts the survival '\n",
      "                                      'benefit of triplet chemotherapy in '\n",
      "                                      'metastatic colorectal cancer}, '\n",
      "                                      'author={Zhang X and Ma H and He Y and '\n",
      "                                      'He W and Chen N and Li Y and Zhong W '\n",
      "                                      'and Wu G and Zhou X and Hua H and Ye F '\n",
      "                                      'and Cai H and Jiang W}, journal={Ther '\n",
      "                                      'Adv Med Oncol}}',\n",
      "                            'NLM': 'Zhang X, Ma H, He Y, He W, Chen N, Li Y, '\n",
      "                                   'Zhong W, Wu G, Zhou X, Hua H, Ye F, Cai H, '\n",
      "                                   'Jiang W. Reclassification of RAS/BRAF '\n",
      "                                   'allele mutations predicts the survival '\n",
      "                                   'benefit of triplet chemotherapy in '\n",
      "                                   'metastatic colorectal cancer Ther Adv Med '\n",
      "                                   'Oncol. PMID: 38293276'},\n",
      "              'date': '2024-01-28T00:00:00Z',\n",
      "              'is_sup_mat_match': False,\n",
      "              'journal': 'Ther Adv Med Oncol',\n",
      "              'pmcid': 'PMC10826380',\n",
      "              'pmid': 38293276,\n",
      "              'snippets': [{'_id': '38293276@15076',\n",
      "                            'mentions': ['106|4|variant|litvar@rs121913530##&&&litvar@#3845#&&&litvar@CA122528#rs121913530##|G12C',\n",
      "                                         '83|4|gene|ncbi_gene@3845|KRAS',\n",
      "                                         '69|4|variant|litvar@rs121913529##&&&litvar@#3845#&&&litvar@CA122540#rs121913529##|G12V',\n",
      "                                         '50|4|variant|litvar@rs121913529##&&&litvar@#3845#&&&litvar@CA122538#rs121913529##|G12D',\n",
      "                                         '123|4|gene|ncbi_gene@3845|KRAS',\n",
      "                                         '128|5|variant|litvar@rs121913527##&&&litvar@#3845#&&&litvar@CA245262#rs121913527##|A146T',\n",
      "                                         '45|4|gene|ncbi_gene@3845|KRAS',\n",
      "                                         '88|4|variant|litvar@rs112445441##&&&litvar@#3845#&&&litvar@CA122534#rs112445441##|G13D',\n",
      "                                         '101|4|gene|ncbi_gene@3845|KRAS',\n",
      "                                         '19|4|gene|ncbi_gene@3845|KRAS',\n",
      "                                         '64|4|gene|ncbi_gene@3845|KRAS'],\n",
      "                            'section': 'results',\n",
      "                            'snip_text': 'The most prevalent KRAS-mutated '\n",
      "                                         'alleles were KRAS G12D (14.7%), KRAS '\n",
      "                                         'G12V (10.7%), KRAS G13D (9.8%), KRAS '\n",
      "                                         'G12C (3.2%), and KRAS A146T (2.0%).',\n",
      "                            'text_hl': 'The most prevalent KRAS-mutated '\n",
      "                                       'alleles were KRAS G12D (14.7%), KRAS '\n",
      "                                       'G12V (10.7%), KRAS G13D (9.8%), KRAS '\n",
      "                                       'G12C (3.2%), and KRAS <m>A146T</m> '\n",
      "                                       '(2.0%).'}],\n",
      "              'title': 'Reclassification of RAS/BRAF allele mutations predicts '\n",
      "                       'the survival benefit of triplet chemotherapy in '\n",
      "                       'metastatic colorectal cancer',\n",
      "              'year': 2024},\n",
      "             {'authors': ['Xu D', 'Shao Q', 'Zhou C', 'Mahmood A', 'Zhang J'],\n",
      "              'citations': {'BibTeX': '@article{37065036, title={In Silico '\n",
      "                                      'Analysis of nsSNPs of Human KRAS Gene '\n",
      "                                      'and Protein Modeling Using '\n",
      "                                      'Bioinformatic Tools}, author={Xu D and '\n",
      "                                      'Shao Q and Zhou C and Mahmood A and '\n",
      "                                      'Zhang J}, journal={ACS Omega}}',\n",
      "                            'NLM': 'Xu D, Shao Q, Zhou C, Mahmood A, Zhang J. '\n",
      "                                   'In Silico Analysis of nsSNPs of Human KRAS '\n",
      "                                   'Gene and Protein Modeling Using '\n",
      "                                   'Bioinformatic Tools ACS Omega. PMID: '\n",
      "                                   '37065036'},\n",
      "              'date': '2023-04-03T00:00:00Z',\n",
      "              'is_sup_mat_match': False,\n",
      "              'journal': 'ACS Omega',\n",
      "              'pmcid': 'PMC10099408',\n",
      "              'pmid': 37065036,\n",
      "              'snippets': [{'_id': '37065036@982',\n",
      "                            'mentions': ['20|35|variant|litvar@rs121913527##&&&litvar@#3845#|alanine '\n",
      "                                         'for proline at position 146'],\n",
      "                            'section': 'abstract',\n",
      "                            'snip_text': 'The substitution of alanine for '\n",
      "                                         'proline at position 146 pushed '\n",
      "                                         'several residues toward the center '\n",
      "                                         'of the protein.',\n",
      "                            'text_hl': 'The substitution of alanine for '\n",
      "                                       'proline at position 146 pushed several '\n",
      "                                       'residues toward the center of the '\n",
      "                                       'protein.'}],\n",
      "              'title': 'In Silico Analysis of nsSNPs of Human KRAS Gene and '\n",
      "                       'Protein Modeling Using Bioinformatic Tools',\n",
      "              'year': 2023},\n",
      "             {'authors': ['Allot A',\n",
      "                          'Wei CH',\n",
      "                          'Phan L',\n",
      "                          'Hefferon T',\n",
      "                          'Landrum M',\n",
      "                          'Rehm HL',\n",
      "                          'Lu Z'],\n",
      "              'citations': {'BibTeX': '@article{37268776, title={Tracking '\n",
      "                                      'genetic variants in the biomedical '\n",
      "                                      'literature using LitVar 2.0}, '\n",
      "                                      'author={Allot A and Wei CH and Phan L '\n",
      "                                      'and Hefferon T and Landrum M and Rehm '\n",
      "                                      'HL and Lu Z}, journal={Nat Genet}}',\n",
      "                            'NLM': 'Allot A, Wei CH, Phan L, Hefferon T, '\n",
      "                                   'Landrum M, Rehm HL, Lu Z. Tracking genetic '\n",
      "                                   'variants in the biomedical literature '\n",
      "                                   'using LitVar 2.0 Nat Genet. PMID: '\n",
      "                                   '37268776'},\n",
      "              'date': '2023-06-01T00:00:00Z',\n",
      "              'is_sup_mat_match': False,\n",
      "              'journal': 'Nat Genet',\n",
      "              'pmcid': 'PMC11096795',\n",
      "              'pmid': 37268776,\n",
      "              'snippets': [{'_id': '37268776@755',\n",
      "                            'mentions': ['236|5|variant|litvar@rs121913527##&&&litvar@#3845#&&&litvar@CA245262#rs121913527##|A146T'],\n",
      "                            'section': 'other',\n",
      "                            'snip_text': 'Unfortunately, traditional '\n",
      "                                         'keyword-based search systems such as '\n",
      "                                         'PubMed often yield suboptimal '\n",
      "                                         'results for variant searches because '\n",
      "                                         'of two major challenges: (I) the '\n",
      "                                         'same genetic variant is commonly '\n",
      "                                         'described with different names '\n",
      "                                         \"(e.g., 'A146T' vs.\",\n",
      "                            'text_hl': 'Unfortunately, traditional '\n",
      "                                       'keyword-based search systems such as '\n",
      "                                       'PubMed often yield suboptimal results '\n",
      "                                       'for variant searches because of two '\n",
      "                                       'major challenges: (I) the same genetic '\n",
      "                                       'variant is commonly described with '\n",
      "                                       \"different names (e.g., '<m>A146T</m>' \"\n",
      "                                       'vs.'}],\n",
      "              'title': 'Tracking genetic variants in the biomedical literature '\n",
      "                       'using LitVar 2.0',\n",
      "              'year': 2023},\n",
      "             {'authors': ['Shah SM',\n",
      "                          'Demidova EV',\n",
      "                          'Ringenbach S',\n",
      "                          'Faezov B',\n",
      "                          'Andrake M',\n",
      "                          'Gandhi A',\n",
      "                          'Mur P',\n",
      "                          'Viana-Errasti J',\n",
      "                          'Xiu J',\n",
      "                          'Swensen J',\n",
      "                          'Valle L',\n",
      "                          'Dunbrack RL Jr',\n",
      "                          'Hall MJ',\n",
      "                          'Arora S'],\n",
      "              'citations': {'BibTeX': '@article{38282550, title={Exploring '\n",
      "                                      'Co-occurring POLE Exonuclease and '\n",
      "                                      'Non-exonuclease Domain Mutations and '\n",
      "                                      'Their Impact on Tumor Mutagenicity}, '\n",
      "                                      'author={Shah SM and Demidova EV and '\n",
      "                                      'Ringenbach S and Faezov B and Andrake M '\n",
      "                                      'and Gandhi A and Mur P and '\n",
      "                                      'Viana-Errasti J and Xiu J and Swensen J '\n",
      "                                      'and Valle L and Dunbrack RL Jr and Hall '\n",
      "                                      'MJ and Arora S}, journal={Cancer Res '\n",
      "                                      'Commun}}',\n",
      "                            'NLM': 'Shah SM, Demidova EV, Ringenbach S, Faezov '\n",
      "                                   'B, Andrake M, Gandhi A, Mur P, '\n",
      "                                   'Viana-Errasti J, Xiu J, Swensen J, Valle '\n",
      "                                   'L, Dunbrack RL Jr, Hall MJ, Arora S. '\n",
      "                                   'Exploring Co-occurring POLE Exonuclease '\n",
      "                                   'and Non-exonuclease Domain Mutations and '\n",
      "                                   'Their Impact on Tumor Mutagenicity Cancer '\n",
      "                                   'Res Commun. PMID: 38282550'},\n",
      "              'date': '2024-01-26T00:00:00Z',\n",
      "              'is_sup_mat_match': False,\n",
      "              'journal': 'Cancer Res Commun',\n",
      "              'pmcid': 'PMC10812383',\n",
      "              'pmid': 38282550,\n",
      "              'snippets': [{'_id': '38282550@30566',\n",
      "                            'mentions': ['58|4|gene|ncbi_gene@3845|KRAS',\n",
      "                                         '18|5|variant|litvar@rs121913527##&&&litvar@#3845#&&&litvar@CA245262#rs121913527##|A146T',\n",
      "                                         '13|4|gene|ncbi_gene@3845|KRAS',\n",
      "                                         '95|4|gene|ncbi_gene@3845|KRAS'],\n",
      "                            'section': 'results',\n",
      "                            'snip_text': 'In contrast, KRAS A146T:overall, a '\n",
      "                                         'less commonly observed KRAS mutation '\n",
      "                                         ':was the most frequent KRAS mutation '\n",
      "                                         'in Groups 2 and 3, and not found in '\n",
      "                                         'Groups 1 and 4.',\n",
      "                            'text_hl': 'In contrast, KRAS A146T:overall, a '\n",
      "                                       'less commonly observed KRAS mutation '\n",
      "                                       ':was the most frequent KRAS mutation '\n",
      "                                       'in Groups 2 and 3, and not found in '\n",
      "                                       'Groups 1 and 4.'}],\n",
      "              'title': 'Exploring Co-occurring POLE Exonuclease and '\n",
      "                       'Non-exonuclease Domain Mutations and Their Impact on '\n",
      "                       'Tumor Mutagenicity',\n",
      "              'year': 2024},\n",
      "             {'authors': ['Chang L',\n",
      "                          'Che G',\n",
      "                          'Yang Q',\n",
      "                          'Lai S',\n",
      "                          'Teng J',\n",
      "                          'Duan J',\n",
      "                          'Liu T',\n",
      "                          'Liu F'],\n",
      "              'citations': {'BibTeX': '@article{37435524, title={Leishmania '\n",
      "                                      'donovani visceral leishmaniasis '\n",
      "                                      'diagnosed by metagenomics '\n",
      "                                      'next-generation sequencing in an infant '\n",
      "                                      'with acute lymphoblastic leukemia: a '\n",
      "                                      'case report}, author={Chang L and Che G '\n",
      "                                      'and Yang Q and Lai S and Teng J and '\n",
      "                                      'Duan J and Liu T and Liu F}, '\n",
      "                                      'journal={Front Public Health}}',\n",
      "                            'NLM': 'Chang L, Che G, Yang Q, Lai S, Teng J, '\n",
      "                                   'Duan J, Liu T, Liu F. Leishmania donovani '\n",
      "                                   'visceral leishmaniasis diagnosed by '\n",
      "                                   'metagenomics next-generation sequencing in '\n",
      "                                   'an infant with acute lymphoblastic '\n",
      "                                   'leukemia: a case report Front Public '\n",
      "                                   'Health. PMID: 37435524'},\n",
      "              'date': '2023-06-22T00:00:00Z',\n",
      "              'is_sup_mat_match': False,\n",
      "              'journal': 'Front Public Health',\n",
      "              'pmcid': 'PMC10332309',\n",
      "              'pmid': 37435524,\n",
      "              'snippets': [{'_id': '37435524@4193',\n",
      "                            'mentions': ['135|4|gene|ncbi_gene@3845|KRAS',\n",
      "                                         '150|4|gene|ncbi_gene@3845|KRAS',\n",
      "                                         '140|5|variant|litvar@rs1057519725##&&&litvar@#3845#|A146V',\n",
      "                                         '108|4|gene|ncbi_gene@5830|PEX5',\n",
      "                                         '103|4|gene|ncbi_gene@2120|ETV6',\n",
      "                                         '155|5|variant|litvar@rs121913527##&&&litvar@#3845#&&&litvar@CA245262#rs121913527##|A146T'],\n",
      "                            'section': 'other',\n",
      "                            'snip_text': 'Based on the above testing, the boy '\n",
      "                                         'was diagnosed with B-cell acute '\n",
      "                                         'lymphoblastic leukemia (B-ALL, L2, '\n",
      "                                         'ETV6-PEX5 fusion gene positive, KRAS '\n",
      "                                         'A146V and KRAS A146T mutation, '\n",
      "                                         'IKZF1-8 heterozygous deletion, 45, '\n",
      "                                         'XY, der (7; 12) (q10; q10)(5)/46, XY '\n",
      "                                         '(15).',\n",
      "                            'text_hl': 'Based on the above testing, the boy '\n",
      "                                       'was diagnosed with B-cell acute '\n",
      "                                       'lymphoblastic leukemia (B-ALL, L2, '\n",
      "                                       'ETV6-PEX5 fusion gene positive, KRAS '\n",
      "                                       'A146V and KRAS <m>A146T</m> mutation, '\n",
      "                                       'IKZF1-8 heterozygous deletion, 45, XY, '\n",
      "                                       'der (7; 12) (q10; q10)(5)/46, XY '\n",
      "                                       '(15).'}],\n",
      "              'title': 'Leishmania donovani visceral leishmaniasis diagnosed '\n",
      "                       'by metagenomics next-generation sequencing in an '\n",
      "                       'infant with acute lymphoblastic leukemia: a case '\n",
      "                       'report',\n",
      "              'year': 2023},\n",
      "             {'authors': ['Li D',\n",
      "                          'Sheppard SE',\n",
      "                          'March ME',\n",
      "                          'Battig MR',\n",
      "                          'Surrey LF',\n",
      "                          'Srinivasan AS',\n",
      "                          'Matsuoka LS',\n",
      "                          'Tian L',\n",
      "                          'Wang F',\n",
      "                          'Seiler C',\n",
      "                          'Dayneka J',\n",
      "                          'Borst AJ',\n",
      "                          'Matos MC',\n",
      "                          'Paulissen SM',\n",
      "                          'Krishnamurthy G',\n",
      "                          'Nriagu B',\n",
      "                          'Sikder T',\n",
      "                          'Casey M',\n",
      "                          'Williams L',\n",
      "                          'Rangu S',\n",
      "                          \"O'Connor N\",\n",
      "                          'Thomas A',\n",
      "                          'Pinto E',\n",
      "                          'Hou C',\n",
      "                          'Nguyen K',\n",
      "                          'Pellegrino da Silva R',\n",
      "                          'Chehimi SN',\n",
      "                          'Kao C',\n",
      "                          'Biroc L',\n",
      "                          'Britt AD',\n",
      "                          'Queenan M',\n",
      "                          'Reid JR',\n",
      "                          'Napoli JA',\n",
      "                          'Low DM',\n",
      "                          'Vatsky S',\n",
      "                          'Treat J',\n",
      "                          'Smith CL',\n",
      "                          'Cahill AM',\n",
      "                          'Snyder KM',\n",
      "                          'Adams DM',\n",
      "                          'Dori Y',\n",
      "                          'Hakonarson H'],\n",
      "              'citations': {'BibTeX': '@article{37264205, title={Genomic '\n",
      "                                      'profiling informs diagnoses and '\n",
      "                                      'treatment in vascular anomalies}, '\n",
      "                                      'author={Li D and Sheppard SE and March '\n",
      "                                      'ME and Battig MR and Surrey LF and '\n",
      "                                      'Srinivasan AS and Matsuoka LS and Tian '\n",
      "                                      'L and Wang F and Seiler C and Dayneka J '\n",
      "                                      'and Borst AJ and Matos MC and Paulissen '\n",
      "                                      'SM and Krishnamurthy G and Nriagu B and '\n",
      "                                      'Sikder T and Casey M and Williams L and '\n",
      "                                      \"Rangu S and O'Connor N and Thomas A and \"\n",
      "                                      'Pinto E and Hou C and Nguyen K and '\n",
      "                                      'Pellegrino da Silva R and Chehimi SN '\n",
      "                                      'and Kao C and Biroc L and Britt AD and '\n",
      "                                      'Queenan M and Reid JR and Napoli JA and '\n",
      "                                      'Low DM and Vatsky S and Treat J and '\n",
      "                                      'Smith CL and Cahill AM and Snyder KM '\n",
      "                                      'and Adams DM and Dori Y and Hakonarson '\n",
      "                                      'H}, journal={Nat Med}}',\n",
      "                            'NLM': 'Li D, Sheppard SE, March ME, Battig MR, '\n",
      "                                   'Surrey LF, Srinivasan AS, Matsuoka LS, '\n",
      "                                   'Tian L, Wang F, Seiler C, Dayneka J, Borst '\n",
      "                                   'AJ, Matos MC, Paulissen SM, Krishnamurthy '\n",
      "                                   'G, Nriagu B, Sikder T, Casey M, Williams '\n",
      "                                   \"L, Rangu S, O'Connor N, Thomas A, Pinto E, \"\n",
      "                                   'Hou C, Nguyen K, Pellegrino da Silva R, '\n",
      "                                   'Chehimi SN, Kao C, Biroc L, Britt AD, '\n",
      "                                   'Queenan M, Reid JR, Napoli JA, Low DM, '\n",
      "                                   'Vatsky S, Treat J, Smith CL, Cahill AM, '\n",
      "                                   'Snyder KM, Adams DM, Dori Y, Hakonarson H. '\n",
      "                                   'Genomic profiling informs diagnoses and '\n",
      "                                   'treatment in vascular anomalies Nat Med. '\n",
      "                                   'PMID: 37264205'},\n",
      "              'date': '2023-06-01T00:00:00Z',\n",
      "              'is_sup_mat_match': False,\n",
      "              'journal': 'Nat Med',\n",
      "              'pmcid': 'PMC11184491',\n",
      "              'pmid': 37264205,\n",
      "              'snippets': [{'_id': '37264205@19103',\n",
      "                            'mentions': ['31|5|variant|litvar@rs121913527##&&&litvar@#3845#&&&litvar@CA245262#rs121913527##|A146T',\n",
      "                                         '26|4|gene|ncbi_gene@445289|KRAS',\n",
      "                                         '40|9|species|ncbi_taxonomy@7955|zebrafish'],\n",
      "                            'section': 'results',\n",
      "                            'snip_text': 'Comparably, expression of KRAS-A146T '\n",
      "                                         'in zebrafish resulted in thoracic '\n",
      "                                         'duct dilatation and tissue expansion '\n",
      "                                         '(Extended Data Fig. 1e-g).',\n",
      "                            'text_hl': 'Comparably, expression of '\n",
      "                                       'KRAS-<m>A146T</m> in zebrafish '\n",
      "                                       'resulted in thoracic duct dilatation '\n",
      "                                       'and tissue expansion (Extended Data '\n",
      "                                       'Fig. 1e-g).'}],\n",
      "              'title': 'Genomic profiling informs diagnoses and treatment in '\n",
      "                       'vascular anomalies',\n",
      "              'year': 2023},\n",
      "             {'authors': ['He F', 'He X', 'Cui M', 'Wang Y'],\n",
      "              'citations': {'BibTeX': '@article{36849965, title={Mixed small '\n",
      "                                      'cell neuroendocrine carcinoma and '\n",
      "                                      'squamous cell carcinoma covered by '\n",
      "                                      'tubulovillous adenoma in the rectum: a '\n",
      "                                      'case report and detailed molecular '\n",
      "                                      'analyses.}, author={He F and He X and '\n",
      "                                      'Cui M and Wang Y}, journal={World J '\n",
      "                                      'Surg Oncol}}',\n",
      "                            'NLM': 'He F, He X, Cui M, Wang Y. Mixed small '\n",
      "                                   'cell neuroendocrine carcinoma and squamous '\n",
      "                                   'cell carcinoma covered by tubulovillous '\n",
      "                                   'adenoma in the rectum: a case report and '\n",
      "                                   'detailed molecular analyses. World J Surg '\n",
      "                                   'Oncol. PMID: 36849965'},\n",
      "              'date': '2023-02-27T00:00:00Z',\n",
      "              'is_sup_mat_match': False,\n",
      "              'journal': 'World J Surg Oncol',\n",
      "              'pmcid': 'PMC9969638',\n",
      "              'pmid': 36849965,\n",
      "              'snippets': [{'_id': '36849965@6966',\n",
      "                            'figure_id': 'Tab1',\n",
      "                            'mentions': ['474|6|variant|litvar@rs1044896377##&&&litvar@#54894#&&&litvar@CA292015224#rs1044896377##|p.R25Q',\n",
      "                                         '185|6|variant|litvar@#7157#p.53dup&&&litvar@#7157#|p.W53*',\n",
      "                                         '96|11|variant|litvar@rs397509212##&&&litvar@#672#&&&litvar@CA3132#rs397509212##|c.4987-2A>G',\n",
      "                                         '268|6|variant|litvar@#55294#p.73dup&&&litvar@#55294#|p.Q73*',\n",
      "                                         '508|7|variant|litvar@rs188493164##&&&litvar@#27030#&&&litvar@CA7276018#rs188493164##|p.M162V',\n",
      "                                         '388|10|variant|litvar@#54894#p.Q44RfsX7&&&litvar@#54894#|p.Q44Rfs*7',\n",
      "                                         '574|5|gene|ncbi_gene@5314|PKHD1',\n",
      "                                         '592|8|variant|litvar@rs554782195##&&&litvar@#5314#&&&litvar@CA3850973#rs554782195##|p.S3669L',\n",
      "                                         '225|8|variant|litvar@#324#p.E1322*&&&litvar@#324#|p.E1322*',\n",
      "                                         '49|4|gene|ncbi_gene@3845|KRAS',\n",
      "                                         '459|5|gene|ncbi_gene@9757|KMT2B',\n",
      "                                         '372|5|gene|ncbi_gene@54894|RNF43',\n",
      "                                         '550|7|variant|litvar@rs764858967##&&&litvar@#5159#&&&litvar@CA3508167#rs764858967##|p.S296C',\n",
      "                                         '342|12|variant|litvar@rs397508924##&&&litvar@#672#&&&litvar@CA327793#rs397508924##|p.Y655Lfs*13',\n",
      "                                         '175|8|variant|litvar@rs1064794618##&&&litvar@#7157#|c.159G>A',\n",
      "                                         '214|9|variant|litvar@rs764773634##&&&litvar@#9757#&&&litvar@CA9384927#rs764773634##|c.3964G>T',\n",
      "                                         '417|4|gene|ncbi_gene@2324|FLT4',\n",
      "                                         '64|7|variant|litvar@rs121913527##&&&litvar@#3845#&&&litvar@CA245262#rs121913527##|p.A146T',\n",
      "                                         '258|8|variant|litvar@rs1057520008##&&&litvar@#7157#|c.217C>T',\n",
      "                                         '210|3|gene|ncbi_gene@324|APC',\n",
      "                                         '170|4|gene|ncbi_gene@7157|TP53',\n",
      "                                         '378|8|variant|litvar@#54894#c.131del&&&litvar@#54894#|c.131del',\n",
      "                                         '498|8|variant|litvar@rs188493164##&&&litvar@#27030#&&&litvar@CA7276018#rs188493164##|c.484A>G',\n",
      "                                         '31|15|disease|ncbi_mesh@D002294|Adenoma '\n",
      "                                         'NEC SCC',\n",
      "                                         '299|10|variant|litvar@#55294#c.372+1G>A&&&litvar@#55294#|c.372+1G>A',\n",
      "                                         '465|7|variant|litvar@rs1407730273##&&&litvar@#5159#|c.74G>A',\n",
      "                                         '54|8|variant|litvar@rs121913527##&&&litvar@#3845#&&&litvar@CA245262#rs121913527##|c.436G>A',\n",
      "                                         '90|5|gene|ncbi_gene@672|BRCA1',\n",
      "                                         '125|4|gene|ncbi_gene@1019|CDK4',\n",
      "                                         '540|8|variant|litvar@rs764858967##&&&litvar@#5159#&&&litvar@CA3508167#rs764858967##|c.886A>T',\n",
      "                                         '422|9|variant|litvar@rs756924526##&&&litvar@#2324#&&&litvar@CA3606728#rs756924526##|c.1426C>T',\n",
      "                                         '433|7|variant|litvar@rs756924526##&&&litvar@#2324#&&&litvar@CA3606728#rs756924526##|p.R476W',\n",
      "                                         '331|9|variant|litvar@rs397508924##&&&litvar@#672#|c.1963dup',\n",
      "                                         '580|10|variant|litvar@rs554782195##&&&litvar@#5314#&&&litvar@CA3850973#rs554782195##|c.11006C>T',\n",
      "                                         '293|5|gene|ncbi_gene@55294|FBXW7',\n",
      "                                         '493|4|gene|ncbi_gene@27030|MLH3',\n",
      "                                         '533|6|gene|ncbi_gene@5159|PDGFRB',\n",
      "                                         '252|5|gene|ncbi_gene@55294|FBXW7',\n",
      "                                         '327|3|gene|ncbi_gene@5925|RB1'],\n",
      "                            'section': 'table',\n",
      "                            'snip_text': 'Gene Mutations Abundance (%)   '\n",
      "                                         'Adenoma NEC SCC   KRAS c.436G>A '\n",
      "                                         '(p.A146T) 25.5 28.3 31.7   BRCA1 '\n",
      "                                         'c.4987-2A>G 45.3 34.2 27.7   CDK4 '\n",
      "                                         'Copy number amplification - CN: 4.6 '\n",
      "                                         '-   TP53 c.159G>A (p.W53*) 41.6 85.2 '\n",
      "                                         '79.8   APC c.3964G>T (p.E1322*) 52.3 '\n",
      "                                         '94.0 78.3   FBXW7 c.217C>T (p.Q73*) '\n",
      "                                         '27.6 47.5 37.3   FBXW7 c.372+1G>A '\n",
      "                                         '26.4 44.2 32.4   RB1 c.1963dup '\n",
      "                                         '(p.Y655Lfs*13) - 67.3% 71.2%   RNF43 '\n",
      "                                         'c.131del (p.Q44Rfs*7) 25.5 52.1 '\n",
      "                                         '53.6   FLT4 c.1426C>T (p.R476W) 28.9 '\n",
      "                                         '49.6 52.5   KMT2B c.74G>A (p.R25Q) - '\n",
      "                                         '- 22.7   MLH3 c.484A>G (p.M162V) '\n",
      "                                         '49.4 7.3 14.3   PDGFRB c.886A>T '\n",
      "                                         '(p.S296C) 22.2 6.2 8.8   PKHD1 '\n",
      "                                         'c.11006C>T (p.S3669L) 26.3 60.9 '\n",
      "                                         '53.1   TMB Mutations/Mbases 6.2 8.2 '\n",
      "                                         '10.3   Low < 10 Low Low High   High '\n",
      "                                         '>=10      Microsatellite  Stable '\n",
      "                                         'Stable Stable   ',\n",
      "                            'text_hl': 'Gene Mutations Abundance (%)   Adenoma '\n",
      "                                       'NEC SCC   KRAS c.436G>A '\n",
      "                                       '(<m>p.A146T</m>) 25.5 28.3 31.7   '\n",
      "                                       'BRCA1 c.4987-2A>G 45.3 34.2 27.7   '\n",
      "                                       'CDK4 Copy number amplification - CN: '\n",
      "                                       '4.6 -   TP53 c.159G>A (p.W53*) 41.6 '\n",
      "                                       '85.2 79.8   APC c.3964G>T (p.E1322*) '\n",
      "                                       '52.3 94.0 78.3   FBXW7 c.217C>T '\n",
      "                                       '(p.Q73*) 27.6 47.5 37.3   FBXW7 '\n",
      "                                       'c.372+1G>A 26.4 44.2 32.4   RB1 '\n",
      "                                       'c.1963dup (p.Y655Lfs*13) - 67.3% '\n",
      "                                       '71.2%   RNF43 c.131del (p.Q44Rfs*7) '\n",
      "                                       '25.5 52.1 53.6   FLT4 c.1426C>T '\n",
      "                                       '(p.R476W) 28.9 49.6 52.5   KMT2B '\n",
      "                                       'c.74G>A (p.R25Q) - - 22.7   MLH3 '\n",
      "                                       'c.484A>G (p.M162V) 49.4 7.3 14.3   '\n",
      "                                       'PDGFRB c.886A>T (p.S296C) 22.2 6.2 '\n",
      "                                       '8.8   PKHD1 c.11006C>T (p.S3669L) 26.3 '\n",
      "                                       '60.9 53.1   TMB Mutations/Mbases 6.2 '\n",
      "                                       '8.2 10.3   Low < 10 Low Low High   '\n",
      "                                       'High >=10      Microsatellite  Stable '\n",
      "                                       'Stable Stable   '}],\n",
      "              'title': 'Mixed small cell neuroendocrine carcinoma and squamous '\n",
      "                       'cell carcinoma covered by tubulovillous adenoma in the '\n",
      "                       'rectum: a case report and detailed molecular analyses.',\n",
      "              'year': 2023},\n",
      "             {'authors': ['Mahdi Y',\n",
      "                          'Khmou M',\n",
      "                          'Souadka A',\n",
      "                          'Agouri HE',\n",
      "                          'Ech-Charif S',\n",
      "                          'Mounjid C',\n",
      "                          'Khannoussi BE'],\n",
      "              'citations': {'BibTeX': '@article{37277698, title={Correlation '\n",
      "                                      'between KRAS and NRAS mutational status '\n",
      "                                      'and clinicopathological features in 414 '\n",
      "                                      'cases of metastatic colorectal cancer '\n",
      "                                      'in Morocco: the largest North African '\n",
      "                                      'case series}, author={Mahdi Y and Khmou '\n",
      "                                      'M and Souadka A and Agouri HE and '\n",
      "                                      'Ech-Charif S and Mounjid C and '\n",
      "                                      'Khannoussi BE}, journal={BMC '\n",
      "                                      'Gastroenterol}}',\n",
      "                            'NLM': 'Mahdi Y, Khmou M, Souadka A, Agouri HE, '\n",
      "                                   'Ech-Charif S, Mounjid C, Khannoussi BE. '\n",
      "                                   'Correlation between KRAS and NRAS '\n",
      "                                   'mutational status and clinicopathological '\n",
      "                                   'features in 414 cases of metastatic '\n",
      "                                   'colorectal cancer in Morocco: the largest '\n",
      "                                   'North African case series BMC '\n",
      "                                   'Gastroenterol. PMID: 37277698'},\n",
      "              'date': '2023-06-05T00:00:00Z',\n",
      "              'is_sup_mat_match': False,\n",
      "              'journal': 'BMC Gastroenterol',\n",
      "              'pmcid': 'PMC10240782',\n",
      "              'pmid': 37277698,\n",
      "              'snippets': [{'_id': '37277698@8418',\n",
      "                            'figure_id': 'Tab2',\n",
      "                            'mentions': ['305|10|variant|litvar@rs112445441##&&&litvar@#3845#&&&litvar@CA122534#rs112445441##|p.Gly13Asp',\n",
      "                                         '143|10|variant|litvar@rs121913529##&&&litvar@#3845#&&&litvar@CA122538#rs121913529##|p.Gly12Asp',\n",
      "                                         '209|10|variant|litvar@rs121913530##&&&litvar@#3845#&&&litvar@CA135565#rs121913530##|p.Gly12Ser',\n",
      "                                         '347|10|variant|litvar@rs121913527##&&&litvar@#3845#&&&litvar@CA245262#rs121913527##|c.436G '\n",
      "                                         '> A',\n",
      "                                         '132|9|variant|litvar@rs121913529##&&&litvar@#3845#&&&litvar@CA135567#rs121913529##|c.35G '\n",
      "                                         '> A',\n",
      "                                         '527|10|variant|litvar@rs11554290##&&&litvar@#4893#&&&litvar@CA123618#rs11554290##|p.Gln61Arg',\n",
      "                                         '457|5|variant|litvar@#4893#|C > G',\n",
      "                                         '110|9|variant|litvar@rs121913529##&&&litvar@#3845#&&&litvar@CA122540#rs121913529##|c.35G '\n",
      "                                         '> T',\n",
      "                                         '176|9|variant|litvar@rs121913530##&&&litvar@#3845#&&&litvar@CA122528#rs121913530##|c.34G '\n",
      "                                         '> C',\n",
      "                                         '231|10|variant|litvar@rs121913529##&&&litvar@#3845#&&&litvar@CA135567#rs121913529##|p.Gly12Ala',\n",
      "                                         '336|10|variant|litvar@rs121913527##&&&litvar@#3845#|c.436G '\n",
      "                                         '> C',\n",
      "                                         '474|10|variant|litvar@rs104886029##&&&litvar@#3845#&&&litvar@CA269881#rs104886029##|p.Ala59Val',\n",
      "                                         '443|5|variant|litvar@#4893#|C > A',\n",
      "                                         '187|10|variant|litvar@rs121913530##&&&litvar@#3845#&&&litvar@CA122531#rs121913530##|p.Gly12Arg',\n",
      "                                         '370|11|variant|litvar@rs121913527##&&&litvar@#3845#|p.Ala146Pro',\n",
      "                                         '220|9|variant|litvar@rs121913529##&&&litvar@#3845#&&&litvar@CA135567#rs121913529##|c.35G '\n",
      "                                         '> C',\n",
      "                                         '486|10|variant|litvar@rs730880965##&&&litvar@#4893#&&&litvar@CA297023#rs730880965##|p.Ala59Gly',\n",
      "                                         '294|9|variant|litvar@rs112445441##&&&litvar@#3845#&&&litvar@CA122534#rs112445441##|c.38G '\n",
      "                                         '> A',\n",
      "                                         '424|10|variant|litvar@rs730880965##&&&litvar@#4893#&&&litvar@CA297023#rs730880965##|c.175G '\n",
      "                                         '> A',\n",
      "                                         '382|11|variant|litvar@rs121913527##&&&litvar@#3845#&&&litvar@CA245262#rs121913527##|p.Ala146Thr',\n",
      "                                         '463|10|variant|litvar@rs730880965##&&&litvar@#4893#&&&litvar@CA297023#rs730880965##|p.Ala59Thr',\n",
      "                                         '121|10|variant|litvar@rs121913529##&&&litvar@#3845#&&&litvar@CA122540#rs121913529##|p.Gly12Val',\n",
      "                                         '520|5|variant|litvar@#4893#|A > G',\n",
      "                                         '364|5|variant|litvar@#4893#|C > T',\n",
      "                                         '154|9|variant|litvar@rs121913530##&&&litvar@#3845#&&&litvar@CA122528#rs121913530##|c.34G '\n",
      "                                         '> T',\n",
      "                                         '394|11|variant|litvar@rs1057519725##&&&litvar@#3845#|p.Ala146Val',\n",
      "                                         '198|9|variant|litvar@rs121913530##&&&litvar@#3845#&&&litvar@CA135565#rs121913530##|c.34G '\n",
      "                                         '> A',\n",
      "                                         '165|10|variant|litvar@rs121913530##&&&litvar@#3845#&&&litvar@CA122528#rs121913530##|p.Gly12Cys'],\n",
      "                            'section': 'table',\n",
      "                            'snip_text': '\\tN\\t%\\t \\tStatus\\t\\t\\t \\tMutation\\t'\n",
      "                                         '214\\t51.7\\t \\tWild-type\\t193\\t'\n",
      "                                         '46.6\\t \\tUnknow (invalid test)\\t7\\t'\n",
      "                                         '1.7\\t \\tMutation\\t\\t\\t \\tCodon '\n",
      "                                         '12c.35G > T (p.Gly12Val)c.35G > A '\n",
      "                                         '(p.Gly12Asp)c.34G > T '\n",
      "                                         '(p.Gly12Cys)c.34G > C '\n",
      "                                         '(p.Gly12Arg)c.34G > A '\n",
      "                                         '(p.Gly12Ser)c.35G > C (p.Gly12Ala)\\t'\n",
      "                                         '16861542311109\\t'\n",
      "                                         '78.528.525.210.75.14.74.2\\t \\tCodon '\n",
      "                                         '13c.38G > A (p.Gly13Asp)\\t25\\t'\n",
      "                                         '11.7\\t \\tCodon 146c.436G > C/c.436G '\n",
      "                                         '> A/c.437 C > '\n",
      "                                         'T(p.Ala146Pro/p.Ala146Thr/p.Ala146Val)\\t'\n",
      "                                         '16\\t7.5\\t \\tCodon 59c.175G > A / '\n",
      "                                         'c.176 C > A / c.176 C > '\n",
      "                                         'G(p.Ala59Thr/p.Ala59Val/ '\n",
      "                                         'p.Ala59Gly)\\t1\\t0.5\\t \\tCodon '\n",
      "                                         '61c.182 A > G (p.Gln61Arg)\\t4\\t'\n",
      "                                         '1.9\\t \\t',\n",
      "                            'text_hl': '\\tN\\t%\\t \\tStatus\\t\\t\\t \\tMutation\\t'\n",
      "                                       '214\\t51.7\\t \\tWild-type\\t193\\t46.6\\t \\t'\n",
      "                                       'Unknow (invalid test)\\t7\\t1.7\\t \\t'\n",
      "                                       'Mutation\\t\\t\\t \\tCodon 12c.35G > T '\n",
      "                                       '(p.Gly12Val)c.35G > A '\n",
      "                                       '(p.Gly12Asp)c.34G > T '\n",
      "                                       '(p.Gly12Cys)c.34G > C '\n",
      "                                       '(p.Gly12Arg)c.34G > A '\n",
      "                                       '(p.Gly12Ser)c.35G > C (p.Gly12Ala)\\t'\n",
      "                                       '16861542311109\\t'\n",
      "                                       '78.528.525.210.75.14.74.2\\t \\tCodon '\n",
      "                                       '13c.38G > A (p.Gly13Asp)\\t25\\t11.7\\t \\t'\n",
      "                                       'Codon 146c.436G > C/c.436G > A/c.437 C '\n",
      "                                       '> '\n",
      "                                       'T(p.Ala146Pro/<m>p.Ala146Thr</m>/p.Ala146Val)\\t'\n",
      "                                       '16\\t7.5\\t \\tCodon 59c.175G > A / c.176 '\n",
      "                                       'C > A / c.176 C > '\n",
      "                                       'G(p.Ala59Thr/p.Ala59Val/ p.Ala59Gly)\\t'\n",
      "                                       '1\\t0.5\\t \\tCodon 61c.182 A > G '\n",
      "                                       '(p.Gln61Arg)\\t4\\t1.9\\t \\t'}],\n",
      "              'title': 'Correlation between KRAS and NRAS mutational status '\n",
      "                       'and clinicopathological features in 414 cases of '\n",
      "                       'metastatic colorectal cancer in Morocco: the largest '\n",
      "                       'North African case series',\n",
      "              'year': 2023},\n",
      "             {'authors': ['Liao CX',\n",
      "                          'Deng CS',\n",
      "                          'Liang X',\n",
      "                          'Yang JC',\n",
      "                          'Chen ZZ',\n",
      "                          'Lin XY',\n",
      "                          'Lin CF',\n",
      "                          'Chen S',\n",
      "                          'Wu SS'],\n",
      "              'citations': {'BibTeX': '@article{37753076, title={PD-1 blockade '\n",
      "                                      'and radiotherapy combination for '\n",
      "                                      'advanced Epstein-Barr virus-associated '\n",
      "                                      'intrahepatic cholangiocarcinoma: a case '\n",
      "                                      'report and literature review}, '\n",
      "                                      'author={Liao CX and Deng CS and Liang X '\n",
      "                                      'and Yang JC and Chen ZZ and Lin XY and '\n",
      "                                      'Lin CF and Chen S and Wu SS}, '\n",
      "                                      'journal={Front Immunol}}',\n",
      "                            'NLM': 'Liao CX, Deng CS, Liang X, Yang JC, Chen '\n",
      "                                   'ZZ, Lin XY, Lin CF, Chen S, Wu SS. PD-1 '\n",
      "                                   'blockade and radiotherapy combination for '\n",
      "                                   'advanced Epstein-Barr virus-associated '\n",
      "                                   'intrahepatic cholangiocarcinoma: a case '\n",
      "                                   'report and literature review Front '\n",
      "                                   'Immunol. PMID: 37753076'},\n",
      "              'date': '2023-09-11T00:00:00Z',\n",
      "              'is_sup_mat_match': False,\n",
      "              'journal': 'Front Immunol',\n",
      "              'pmcid': 'PMC10518395',\n",
      "              'pmid': 37753076,\n",
      "              'snippets': [{'_id': '37753076@8413',\n",
      "                            'mentions': ['52|10|variant|litvar@#5728#p.D268GfsX&&&litvar@#5728#|p. '\n",
      "                                         'D268Gfs',\n",
      "                                         '47|4|gene|ncbi_gene@5728|PTEN',\n",
      "                                         '78|4|gene|ncbi_gene@3845|KRAS',\n",
      "                                         '83|7|variant|litvar@rs121913527##&&&litvar@#3845#&&&litvar@CA245262#rs121913527##|p.A146T'],\n",
      "                            'section': 'other',\n",
      "                            'snip_text': 'The genomic profiling revealed the '\n",
      "                                         'presence of PTEN p. D268Gfs*30 '\n",
      "                                         '(19.2%) and KRAS p.A146T (18.5%), '\n",
      "                                         'along with high TMB (TMB-H, 53.',\n",
      "                            'text_hl': 'The genomic profiling revealed the '\n",
      "                                       'presence of PTEN p. D268Gfs*30 (19.2%) '\n",
      "                                       'and KRAS <m>p.A146T</m> (18.5%), along '\n",
      "                                       'with high TMB (TMB-H, 53.'}],\n",
      "              'title': 'PD-1 blockade and radiotherapy combination for '\n",
      "                       'advanced Epstein-Barr virus-associated intrahepatic '\n",
      "                       'cholangiocarcinoma: a case report and literature '\n",
      "                       'review',\n",
      "              'year': 2023},\n",
      "             {'authors': ['Sagawa T',\n",
      "                          'Sato Y',\n",
      "                          'Hirakawa M',\n",
      "                          'Hamaguchi K',\n",
      "                          'Tamura F',\n",
      "                          'Nagashima H',\n",
      "                          'Fujikawa K',\n",
      "                          'Okamoto K',\n",
      "                          'Kawano Y',\n",
      "                          'Sogabe M',\n",
      "                          'Miyamoto H',\n",
      "                          'Takayama T'],\n",
      "              'citations': {'BibTeX': '@article{37434969, title={Case Report: '\n",
      "                                      'Longitudinal monitoring of clonal '\n",
      "                                      'evolution by circulating tumor DNA for '\n",
      "                                      'resistance to anti-EGFR antibody in a '\n",
      "                                      'case of metastatic colorectal cancer}, '\n",
      "                                      'author={Sagawa T and Sato Y and '\n",
      "                                      'Hirakawa M and Hamaguchi K and Tamura F '\n",
      "                                      'and Nagashima H and Fujikawa K and '\n",
      "                                      'Okamoto K and Kawano Y and Sogabe M and '\n",
      "                                      'Miyamoto H and Takayama T}, '\n",
      "                                      'journal={Front Oncol}}',\n",
      "                            'NLM': 'Sagawa T, Sato Y, Hirakawa M, Hamaguchi K, '\n",
      "                                   'Tamura F, Nagashima H, Fujikawa K, Okamoto '\n",
      "                                   'K, Kawano Y, Sogabe M, Miyamoto H, '\n",
      "                                   'Takayama T. Case Report: Longitudinal '\n",
      "                                   'monitoring of clonal evolution by '\n",
      "                                   'circulating tumor DNA for resistance to '\n",
      "                                   'anti-EGFR antibody in a case of metastatic '\n",
      "                                   'colorectal cancer Front Oncol. PMID: '\n",
      "                                   '37434969'},\n",
      "              'date': '2023-06-26T00:00:00Z',\n",
      "              'is_sup_mat_match': False,\n",
      "              'journal': 'Front Oncol',\n",
      "              'pmcid': 'PMC10332633',\n",
      "              'pmid': 37434969,\n",
      "              'snippets': [{'_id': '37434969@8912',\n",
      "                            'mentions': ['257|5|variant|litvar@rs121913527##&&&litvar@#3845#&&&litvar@CA245262#rs121913527##|A146T',\n",
      "                                         '186|4|variant|litvar@rs121913254##&&&litvar@#4893#&&&litvar@CA151263#rs121913254##|Q61K',\n",
      "                                         '122|4|variant|litvar@rs121434595##&&&litvar@#4893#&&&litvar@CA151261#rs121434595##|G13R',\n",
      "                                         '275|5|variant|litvar@rs1057519725##&&&litvar@#3845#|A146V',\n",
      "                                         '198|4|variant|litvar@rs11554290##&&&litvar@#4893#&&&litvar@CA16602361#rs11554290##|Q61L',\n",
      "                                         '79|4|variant|litvar@rs121913530##&&&litvar@#3845#&&&litvar@CA122528#rs121913530##|G12C',\n",
      "                                         '297|4|gene|ncbi_gene@673|BRAF',\n",
      "                                         '264|5|variant|litvar@rs121913527##&&&litvar@#3845#|A146P',\n",
      "                                         '210|4|variant|litvar@rs11554290##&&&litvar@#4893#&&&litvar@CA123618#rs11554290##|Q61R',\n",
      "                                         '250|5|variant|litvar@rs770248150##&&&litvar@#3845#&&&litvar@CA6486888#rs770248150##|K117N',\n",
      "                                         '104|4|variant|litvar@rs1391410075##&&&litvar@#2064#|G12A',\n",
      "                                         '116|4|variant|litvar@rs774487133##&&&litvar@#1956#&&&litvar@CA4265099#rs774487133##|G13C',\n",
      "                                         '180|4|variant|litvar@rs781085650##&&&litvar@#673#&&&litvar@CA4517046#rs781085650##|A59G',\n",
      "                                         '91|4|variant|litvar@rs121913529##&&&litvar@#3845#&&&litvar@CA122538#rs121913529##|G12D',\n",
      "                                         '134|4|variant|litvar@rs121434596##&&&litvar@#4893#&&&litvar@CA16602363#rs121434596##|G13V',\n",
      "                                         '311|5|variant|litvar@rs113488022##&&&litvar@#673#&&&litvar@CA123643#rs113488022##|V600E',\n",
      "                                         '110|4|variant|litvar@rs564064363##&&&litvar@#2064#&&&litvar@CA8533531#rs564064363##|G13S',\n",
      "                                         '128|4|variant|litvar@rs112445441##&&&litvar@#3845#&&&litvar@CA122534#rs112445441##|G13D',\n",
      "                                         '174|4|variant|litvar@rs730880965##&&&litvar@#4893#&&&litvar@CA297023#rs730880965##|A59T',\n",
      "                                         '97|5|variant|litvar@rs121913529##&&&litvar@#3845#&&&litvar@CA122540#rs121913529##|G12 '\n",
      "                                         'V',\n",
      "                                         '144|4|variant|litvar@rs768282987##&&&litvar@#2064#&&&litvar@CA290415598#rs768282987##|G13A',\n",
      "                                         '192|4|variant|litvar@rs766186178##&&&litvar@#2064#&&&litvar@CA8533544#rs766186178##|Q61E',\n",
      "                                         '73|4|variant|litvar@rs121913530##&&&litvar@#3845#&&&litvar@CA135565#rs121913530##|G12S',\n",
      "                                         '220|4|variant|litvar@rs17851045##&&&litvar@#3845#&&&litvar@CA135552#rs17851045##|Q61H',\n",
      "                                         '204|4|variant|litvar@rs11554290##&&&litvar@#4893#&&&litvar@CA10602732#rs11554290##|Q61P',\n",
      "                                         '85|4|variant|litvar@rs121913530##&&&litvar@#3845#&&&litvar@CA122531#rs121913530##|G12R'],\n",
      "                            'section': 'other',\n",
      "                            'snip_text': 'Nagoya, Japan), which can '\n",
      "                                         'simultaneously examine 12 types of '\n",
      "                                         'RAS exon 2 (G12S, G12C, G12R, G12D, '\n",
      "                                         'G12 V, G12A, G13S, G13C, G13R, G13D, '\n",
      "                                         'G13V, and G13A), 8 types of RAS exon '\n",
      "                                         '3 (A59T, A59G, Q61K, Q61E, Q61L, '\n",
      "                                         'Q61P, Q61R, and Q61H), 4 types of '\n",
      "                                         'RAS exon 4 (K117N, A146T, A146P, and '\n",
      "                                         'A146V) mutations, and BRAF exon 15 '\n",
      "                                         '(V600E).',\n",
      "                            'text_hl': 'Nagoya, Japan), which can '\n",
      "                                       'simultaneously examine 12 types of RAS '\n",
      "                                       'exon 2 (G12S, G12C, G12R, G12D, G12 V, '\n",
      "                                       'G12A, G13S, G13C, G13R, G13D, G13V, '\n",
      "                                       'and G13A), 8 types of RAS exon 3 '\n",
      "                                       '(A59T, A59G, Q61K, Q61E, Q61L, Q61P, '\n",
      "                                       'Q61R, and Q61H), 4 types of RAS exon 4 '\n",
      "                                       '(K117N, <m>A146T</m>, A146P, and '\n",
      "                                       'A146V) mutations, and BRAF exon 15 '\n",
      "                                       '(V600E).'}],\n",
      "              'title': 'Case Report: Longitudinal monitoring of clonal '\n",
      "                       'evolution by circulating tumor DNA for resistance to '\n",
      "                       'anti-EGFR antibody in a case of metastatic colorectal '\n",
      "                       'cancer',\n",
      "              'year': 2023}],\n",
      " 'total_pages': 78}\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "import pprint\n",
    "def search_litvar(variant_id):\n",
    "    \"\"\"\n",
    "    NCBI LitVar API   variant_id   .\n",
    "\n",
    "    Parameters:\n",
    "        variant_id (str):    ID (: 'rs121913527')\n",
    "\n",
    "    Returns:\n",
    "        dict: API  JSON    \n",
    "    \"\"\"\n",
    "    base_url = 'https://www.ncbi.nlm.nih.gov/research/litvar2-api/search/'\n",
    "    params = {\n",
    "        'variant': f'litvar@{variant_id}##',\n",
    "        'filters': '{}',\n",
    "        'sort': 'score desc',\n",
    "        'page': 2,\n",
    "    }\n",
    "\n",
    "    try:\n",
    "        response = requests.get(base_url, params=params)\n",
    "        response.raise_for_status()  # HTTP    \n",
    "        return response.json()\n",
    "    except requests.exceptions.HTTPError as errh:\n",
    "        print(\"HTTP :\", errh)\n",
    "    except requests.exceptions.ConnectionError as errc:\n",
    "        print(\" :\", errc)\n",
    "    except requests.exceptions.Timeout as errt:\n",
    "        print(\"  :\", errt)\n",
    "    except requests.exceptions.RequestException as err:\n",
    "        print(\" :\", err)\n",
    "\n",
    "#  \n",
    "if __name__ == \"__main__\":\n",
    "    variant_id = 'rs121913527'\n",
    "    result = search_litvar(variant_id)\n",
    "    pprint.pprint(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "78 1 10\n",
      "78 2 10\n",
      "78 3 10\n",
      "78 4 10\n",
      "78 5 10\n",
      "78 6 10\n",
      "78 7 10\n",
      "78 8 10\n",
      "78 9 10\n",
      "78 10 10\n",
      "78 11 10\n",
      "78 12 10\n",
      "78 13 10\n",
      "78 14 10\n",
      "78 15 10\n",
      "78 16 10\n",
      "78 17 10\n",
      "78 18 10\n",
      "78 19 10\n",
      "78 20 10\n",
      "78 21 10\n",
      "78 22 10\n",
      "78 23 10\n",
      "78 24 10\n",
      "78 25 10\n",
      "78 26 10\n",
      "78 27 10\n",
      "78 28 10\n",
      "78 29 10\n",
      "78 30 10\n",
      "78 31 10\n",
      "78 32 10\n",
      "78 33 10\n",
      "78 34 10\n",
      "78 35 10\n",
      "78 36 10\n",
      "78 37 10\n",
      "78 38 10\n",
      "78 39 10\n",
      "78 40 10\n",
      "78 41 10\n",
      "78 42 10\n",
      "78 43 10\n",
      "78 44 10\n",
      "78 45 10\n",
      "78 46 10\n",
      "78 47 10\n",
      "78 48 10\n",
      "78 49 10\n",
      "78 50 10\n",
      "78 51 10\n",
      "78 52 10\n",
      "78 53 10\n",
      "78 54 10\n",
      "78 55 10\n",
      "78 56 10\n",
      "78 57 10\n",
      "78 58 10\n",
      "78 59 10\n",
      "78 60 10\n",
      "78 61 10\n",
      "78 62 10\n",
      "78 63 10\n",
      "78 64 10\n",
      "78 65 10\n",
      "78 66 10\n",
      "78 67 10\n",
      "78 68 10\n",
      "78 69 10\n",
      "78 70 10\n",
      "78 71 10\n",
      "78 72 10\n",
      "78 73 10\n",
      "78 74 10\n",
      "78 75 10\n",
      "78 76 10\n",
      "78 77 10\n",
      "78 78 6\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'pprint' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [1]\u001b[0m, in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     55\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;18m__name__\u001b[39m \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m__main__\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[1;32m     56\u001b[0m     variant_id \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mrs121913527\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m---> 57\u001b[0m     results, facets \u001b[38;5;241m=\u001b[39m \u001b[43msearch_litvar_all\u001b[49m\u001b[43m(\u001b[49m\u001b[43mvariant_id\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     58\u001b[0m     \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(results)\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m  .\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "Input \u001b[0;32mIn [1]\u001b[0m, in \u001b[0;36msearch_litvar_all\u001b[0;34m(variant_id)\u001b[0m\n\u001b[1;32m     38\u001b[0m all_results\u001b[38;5;241m.\u001b[39mappend(data[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mresults\u001b[39m\u001b[38;5;124m'\u001b[39m])\n\u001b[1;32m     39\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m total_pages \u001b[38;5;241m==\u001b[39m offset:\n\u001b[0;32m---> 40\u001b[0m     \u001b[43mpprint\u001b[49m\u001b[38;5;241m.\u001b[39mpprint(facets)\n\u001b[1;32m     41\u001b[0m     \u001b[38;5;28;01mbreak\u001b[39;00m\n\u001b[1;32m     42\u001b[0m offset\u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'pprint' is not defined"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "\n",
    "def search_litvar_all(variant_id):\n",
    "    \"\"\"\n",
    "    NCBI LitVar API   variant_id    .\n",
    "\n",
    "    Parameters:\n",
    "        variant_id (str):    ID (: 'rs121913527')\n",
    "\n",
    "    Returns:\n",
    "        list:    \n",
    "    \"\"\"\n",
    "    page = 1\n",
    "    base_url = 'https://www.ncbi.nlm.nih.gov/research/litvar2-api/search/'\n",
    "    params = {\n",
    "        'variant': f'litvar@{variant_id}##',\n",
    "        'filters': '{}',\n",
    "        'sort': 'score desc',\n",
    "        'page': page,  #     (API  )\n",
    "        'page_size': 20\n",
    "    }\n",
    "\n",
    "    all_results = []\n",
    "    offset = 1  #    \n",
    "\n",
    "    try:\n",
    "        while True:\n",
    "            params['page'] = offset\n",
    "            response = requests.get(base_url, params=params)\n",
    "            response.raise_for_status()\n",
    "            data = response.json()\n",
    "            total_pages = data['total_pages']\n",
    "            facets = data['facets']\n",
    "            \n",
    "\n",
    "            #       \n",
    "            print(total_pages, offset, len(data['results']))\n",
    "            all_results.append(data['results'])\n",
    "            if total_pages == offset:\n",
    "                pprint.pprint(facets)\n",
    "                break\n",
    "            offset+=1\n",
    "        return all_results, facets\n",
    "\n",
    "    except requests.exceptions.HTTPError as errh:\n",
    "        print(\"HTTP :\", errh)\n",
    "    except requests.exceptions.ConnectionError as errc:\n",
    "        print(\" :\", errc)\n",
    "    except requests.exceptions.Timeout as errt:\n",
    "        print(\"  :\", errt)\n",
    "    except requests.exceptions.RequestException as err:\n",
    "        print(\" :\", err)\n",
    "\n",
    "#  \n",
    "if __name__ == \"__main__\":\n",
    "    variant_id = 'rs121913527'\n",
    "    results, facets = search_litvar_all(variant_id)\n",
    "    print(f\" {len(results)}  .\")\n",
    "    #      ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_results = sum(results, [])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "77"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test=0\n",
    "for item in new_results:\n",
    "    if item['year']==2018:\n",
    "        test+=1\n",
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_years(results):\n",
    "    return list(set([result['year'] for result in results]))\n",
    "\n",
    "def sort_dict_by_years(results):\n",
    "    years = get_years(results)\n",
    "    set_dict_by_year = {}\n",
    "    for year in years:\n",
    "        result_by_year = [] \n",
    "        for result in results:\n",
    "            if result['year'] == year:\n",
    "                result_by_year.append(result)\n",
    "        set_dict_by_year[year] = result_by_year\n",
    "    return set_dict_by_year \n",
    "                \n",
    "    \n",
    "test_dict=sort_dict_by_years(results=new_results)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "77"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(test_dict[2018])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  : 78\n",
      "{'count': 777,\n",
      " 'journal': {'buckets': [{'count': 59, 'val': 'Oncotarget'},\n",
      "                         {'count': 46, 'val': 'Cancers (Basel)'},\n",
      "                         {'count': 46, 'val': 'PLoS One'},\n",
      "                         {'count': 34, 'val': 'Nat Commun'},\n",
      "                         {'count': 24, 'val': 'Front Oncol'}]},\n",
      " 'section': {'count': 874,\n",
      "             'snippets': {'buckets': [{'count': 276, 'cp': 276, 'val': 'table'},\n",
      "                                      {'count': 215,\n",
      "                                       'cp': 215,\n",
      "                                       'val': 'results'},\n",
      "                                      {'count': 100,\n",
      "                                       'cp': 100,\n",
      "                                       'val': 'discussion'},\n",
      "                                      {'count': 82, 'cp': 82, 'val': 'other'},\n",
      "                                      {'count': 66, 'cp': 66, 'val': 'methods'},\n",
      "                                      {'count': 50,\n",
      "                                       'cp': 50,\n",
      "                                       'val': 'figure_legend'},\n",
      "                                      {'count': 49,\n",
      "                                       'cp': 49,\n",
      "                                       'val': 'abstract'},\n",
      "                                      {'count': 30,\n",
      "                                       'cp': 30,\n",
      "                                       'val': 'introduction'},\n",
      "                                      {'count': 4,\n",
      "                                       'cp': 4,\n",
      "                                       'val': 'conclusion'},\n",
      "                                      {'count': 2, 'cp': 2, 'val': 'title'}]}},\n",
      " 'year': {'buckets': [{'count': 97, 'val': 2022},\n",
      "                      {'count': 85, 'val': 2019},\n",
      "                      {'count': 83, 'val': 2020},\n",
      "                      {'count': 77, 'val': 2018},\n",
      "                      {'count': 76, 'val': 2021},\n",
      "                      {'count': 68, 'val': 2023},\n",
      "                      {'count': 67, 'val': 2017},\n",
      "                      {'count': 57, 'val': 2016},\n",
      "                      {'count': 55, 'val': 2015},\n",
      "                      {'count': 45, 'val': 2024},\n",
      "                      {'count': 30, 'val': 2014},\n",
      "                      {'count': 16, 'val': 2013},\n",
      "                      {'count': 7, 'val': 2012},\n",
      "                      {'count': 5, 'val': 2011},\n",
      "                      {'count': 4, 'val': 2010},\n",
      "                      {'count': 3, 'val': 2009},\n",
      "                      {'count': 1, 'val': 2006},\n",
      "                      {'count': 1, 'val': 2025}]}}\n",
      " 777  .\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "import pprint\n",
    "import concurrent.futures\n",
    "\n",
    "def fetch_page(base_url, params, page):\n",
    "    params['page'] = page\n",
    "    response = requests.get(base_url, params=params)\n",
    "    response.raise_for_status()\n",
    "    return response.json()\n",
    "\n",
    "def search_litvar_all(variant_id):\n",
    "    \"\"\"\n",
    "    NCBI LitVar API   variant_id    .\n",
    "\n",
    "    Parameters:\n",
    "        variant_id (str):    ID (: 'rs121913527')\n",
    "\n",
    "    Returns:\n",
    "        list:    \n",
    "    \"\"\"\n",
    "    base_url = 'https://www.ncbi.nlm.nih.gov/research/litvar2-api/search/'\n",
    "    params = {\n",
    "        'variant': f'litvar@{variant_id}##',\n",
    "        'filters': '{}',\n",
    "        'sort': 'score desc',\n",
    "        'page': 1,  #     (API  )\n",
    "        'page_size': 20\n",
    "    }\n",
    "\n",
    "    try:\n",
    "        #    total_pages \n",
    "        response = requests.get(base_url, params=params)\n",
    "        response.raise_for_status()\n",
    "        data = response.json()\n",
    "        total_pages = data['total_pages']\n",
    "        facets = data['facets']\n",
    "\n",
    "        print(f\"  : {total_pages}\")\n",
    "\n",
    "        all_results = []\n",
    "\n",
    "        # ThreadPoolExecutor     \n",
    "        with concurrent.futures.ThreadPoolExecutor(max_workers=total_pages) as executor:\n",
    "            future_to_page = {executor.submit(fetch_page, base_url, params, page): page for page in range(1, total_pages + 1)}\n",
    "            for future in concurrent.futures.as_completed(future_to_page):\n",
    "                page = future_to_page[future]\n",
    "                try:\n",
    "                    data = future.result()\n",
    "                    all_results.extend(data['results'])\n",
    "                except Exception as exc:\n",
    "                    print(f\"{page}   : {exc}\")\n",
    "\n",
    "        pprint.pprint(facets)\n",
    "        return all_results, facets\n",
    "\n",
    "    except requests.exceptions.HTTPError as errh:\n",
    "        print(\"HTTP :\", errh)\n",
    "    except requests.exceptions.ConnectionError as errc:\n",
    "        print(\" :\", errc)\n",
    "    except requests.exceptions.Timeout as errt:\n",
    "        print(\"  :\", errt)\n",
    "    except requests.exceptions.RequestException as err:\n",
    "        print(\" :\", err)\n",
    "\n",
    "#  \n",
    "if __name__ == \"__main__\":\n",
    "    variant_id = 'rs121913527'\n",
    "    results, facets = search_litvar_all(variant_id)\n",
    "    print(f\" {len(results)}  .\")\n",
    "    #      ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'count': 776,\n",
       " 'journal': {'buckets': [{'val': 'Oncotarget', 'count': 59},\n",
       "   {'val': 'Cancers (Basel)', 'count': 46},\n",
       "   {'val': 'PLoS One', 'count': 46},\n",
       "   {'val': 'Nat Commun', 'count': 34},\n",
       "   {'val': 'Front Oncol', 'count': 24}]},\n",
       " 'year': {'buckets': [{'val': 2022, 'count': 97},\n",
       "   {'val': 2019, 'count': 85},\n",
       "   {'val': 2020, 'count': 83},\n",
       "   {'val': 2018, 'count': 77},\n",
       "   {'val': 2021, 'count': 76},\n",
       "   {'val': 2023, 'count': 68},\n",
       "   {'val': 2017, 'count': 67},\n",
       "   {'val': 2016, 'count': 57},\n",
       "   {'val': 2015, 'count': 55},\n",
       "   {'val': 2024, 'count': 45},\n",
       "   {'val': 2014, 'count': 30},\n",
       "   {'val': 2013, 'count': 16},\n",
       "   {'val': 2012, 'count': 7},\n",
       "   {'val': 2011, 'count': 5},\n",
       "   {'val': 2010, 'count': 4},\n",
       "   {'val': 2009, 'count': 3},\n",
       "   {'val': 2006, 'count': 1}]},\n",
       " 'section': {'count': 871,\n",
       "  'snippets': {'buckets': [{'val': 'table', 'count': 276, 'cp': 276},\n",
       "    {'val': 'results', 'count': 214, 'cp': 214},\n",
       "    {'val': 'discussion', 'count': 99, 'cp': 99},\n",
       "    {'val': 'other', 'count': 81, 'cp': 81},\n",
       "    {'val': 'methods', 'count': 66, 'cp': 66},\n",
       "    {'val': 'figure_legend', 'count': 50, 'cp': 50},\n",
       "    {'val': 'abstract', 'count': 49, 'cp': 49},\n",
       "    {'val': 'introduction', 'count': 30, 'cp': 30},\n",
       "    {'val': 'conclusion', 'count': 4, 'cp': 4},\n",
       "    {'val': 'title', 'count': 2, 'cp': 2}]}}}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "facets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HTTP : 400 Client Error: Bad Request for url: https://www.ncbi.nlm.nih.gov/research/litvar2-api/search/?variant=litvar%40rs1640863258%23%23&filters=%7B%7D&sort=score+desc&page=1&page_size=20\n",
      " 0  .\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "import pprint\n",
    "import concurrent.futures\n",
    "\n",
    "def fetch_page(base_url, params, page):\n",
    "    params['page'] = page\n",
    "    response = requests.get(base_url, params=params)\n",
    "    response.raise_for_status()\n",
    "    return response.json()\n",
    "\n",
    "def search_litvar_all(variant_id):\n",
    "    \"\"\"\n",
    "    NCBI LitVar API   variant_id    .\n",
    "\n",
    "    Parameters:\n",
    "        variant_id (str):    ID (: 'rs121913527')\n",
    "\n",
    "    Returns:\n",
    "        list:    \n",
    "    \"\"\"\n",
    "    base_url = 'https://www.ncbi.nlm.nih.gov/research/litvar2-api/search/'\n",
    "    params = {\n",
    "        'variant': f'litvar@{variant_id}##',\n",
    "        'filters': '{}',\n",
    "        'sort': 'score desc',\n",
    "        'page': 1,  #     (API  )\n",
    "        'page_size': 20\n",
    "    }\n",
    "\n",
    "    try:\n",
    "        #    total_pages \n",
    "        response = requests.get(base_url, params=params)\n",
    "        response.raise_for_status()\n",
    "        data = response.json()\n",
    "        total_pages = data['total_pages']\n",
    "        facets = data['facets']\n",
    "\n",
    "        print(f\"  : {total_pages}\")\n",
    "\n",
    "        all_results = []\n",
    "\n",
    "        # ThreadPoolExecutor     \n",
    "        with concurrent.futures.ThreadPoolExecutor(max_workers=total_pages) as executor:\n",
    "            future_to_page = {executor.submit(fetch_page, base_url, params, page): page for page in range(1, total_pages + 1)}\n",
    "            for future in concurrent.futures.as_completed(future_to_page):\n",
    "                page = future_to_page[future]\n",
    "                try:\n",
    "                    data = future.result()\n",
    "                    all_results.extend(data['results'])\n",
    "                except Exception as exc:\n",
    "                    print(f\"{page}   : {exc}\")\n",
    "\n",
    "        #pprint.pprint(facets)\n",
    "        print(type(all_results))\n",
    "        if all_results == None:\n",
    "            raise ValueError(\"No data in all_results\")\n",
    "        if all_results == []:\n",
    "            raise ValueError(\"No data in all_results\")\n",
    "        return all_results, facets\n",
    "\n",
    "    except requests.exceptions.HTTPError as errh:\n",
    "        print(\"HTTP :\", errh)\n",
    "        return [], []\n",
    "    except requests.exceptions.ConnectionError as errc:\n",
    "        print(\" :\", errc)\n",
    "        return [], []\n",
    "    except requests.exceptions.Timeout as errt:\n",
    "        print(\"  :\", errt)\n",
    "        return [], []\n",
    "    except requests.exceptions.RequestException as err:\n",
    "        print(\" :\", err)\n",
    "        return [], []\n",
    "    except ValueError as errv:\n",
    "        print(\"ValueError:\", errv)\n",
    "        return [], []\n",
    "\n",
    "\n",
    "def response2rsid(response_data):\n",
    "    try:\n",
    "        each_rsid = ''\n",
    "        for response_key in response_data.keys():\n",
    "            each_resource_data = response_data[response_key]\n",
    "            each_data = each_resource_data[0]\n",
    "            if len(each_resource_data) == 0:\n",
    "                raise Exception(\"No data in response_data\")\n",
    "            ex_variants = each_data['ex_variants'].split(\"&\")\n",
    "            #print(ex_variants)\n",
    "            if ((ex_variants[0] == '-') or (ex_variants[0] == '')):\n",
    "                raise Exception('do not support this type of ex_variants')\n",
    "            each_rsid = ''\n",
    "            for item in ex_variants:\n",
    "                if item.startswith('rs'):\n",
    "                    each_rsid = item\n",
    "                    break\n",
    "            #print(each_rsid)\n",
    "        return each_rsid\n",
    "    except Exception as e:\n",
    "        print(e)\n",
    "    \n",
    "def process_litvar(variant, \n",
    "                   host='http://143.248.208.160:4042/', \n",
    "                   route=\"parse2mane\", \n",
    "                   session_data=\"kdw59520@kaist.ac.kr\",\n",
    "                   headers={\"Content-Type\": \"application/json; charset=UTF-8\"}):\n",
    "\n",
    "    data = {\"seq\": variant, \"session_data\": session_data}\n",
    "    response = requests.post(\n",
    "        url=f'{host}{route}',\n",
    "        json=data,\n",
    "        headers=headers)\n",
    "    response_data = response.json()\n",
    "    rsid = response2rsid(response_data=response_data)\n",
    "    results, facets = search_litvar_all(variant_id=rsid)\n",
    "    \n",
    "    print(f\" {len(results)}  .\")\n",
    "    return results, facets\n",
    "\n",
    "#      .\n",
    "#seq=\"chr1:925952-925952 G>A\"\n",
    "#seq=\"chrX:18609570-18609570 G>A\"\n",
    "#seq=\"NM_000368.5(TSC1):c.364-1G>T\"\n",
    "#seq=\"chr3:155116987-155116987 G>A\" #support to cosv id\n",
    "seq=\"chr12:25225628-25225628 C>G\"\n",
    "results, facets = process_litvar(variant=seq)\n",
    "#pprint.pprint(results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "with open('./test.json', 'w') as f:\n",
    "    json.dump(results, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'count': 10,\n",
       " 'journal': {'buckets': [{'val': 'Genet Med', 'count': 2},\n",
       "   {'val': 'Dev Med Child Neurol', 'count': 1},\n",
       "   {'val': 'EMBO J', 'count': 1},\n",
       "   {'val': 'J Med Genet', 'count': 1},\n",
       "   {'val': 'Neural Plast', 'count': 1}]},\n",
       " 'year': {'buckets': [{'val': 2017, 'count': 2},\n",
       "   {'val': 2018, 'count': 2},\n",
       "   {'val': 2021, 'count': 2},\n",
       "   {'val': 2012, 'count': 1},\n",
       "   {'val': 2015, 'count': 1},\n",
       "   {'val': 2016, 'count': 1},\n",
       "   {'val': 2019, 'count': 1}]},\n",
       " 'section': {'count': 9,\n",
       "  'snippets': {'buckets': [{'val': 'table', 'count': 4, 'cp': 4},\n",
       "    {'val': 'figure_legend', 'count': 2, 'cp': 2},\n",
       "    {'val': 'results', 'count': 2, 'cp': 2},\n",
       "    {'val': 'introduction', 'count': 1, 'cp': 1}]}}}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "facets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from variants import Variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['NC_000012.12', 'g.25225628C>A']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "('chr12:25225628-25225628', 'C', 'A')"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Variants(\"rs121913527\").variant2locus()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "portal",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
